### PR TITLE
feature/update-hub-location-sim

### PR DESCRIPTION
**Pull request by edsanville**
Do we want the hub location(s) to persist?  That's another task, I think.

### CODE REVIEW

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (marshalllele) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Is this meant to be commented out? If so, it doesn't look like we need the hubIdSelectionElement() function

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (marshalllele) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: It looks like this function isn't used

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (marshalllele) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Unused import

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (marshalllele) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Unused variable

**Review by marshalllele — CHANGES_REQUESTED** [review]
Functionality works well in sim! Deployed to hub and verified that the Simulation accordion only shows up if in a simulator.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: It is commented out, because as @michael-jaia noted, the sim doesn't allow multi-hub yet.  I kept the function around in case we want it someday.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: (see above)

**Review by marshalllele — DISMISSED** [review]
Everything looks good! Testing in sim and deployed to a hub to make sure the Simulation panel only appears in a simulation.

File: src/lib/messages/jaia_dccl.proto
Comment (michael-jaia) [inline src/lib/messages/jaia_dccl.proto]: Could `set_hub_location` be confusing the way protocol buffers generates setters. Would `hub_location` work here?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can these eslint lines be removed?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we make this `hubID` or `hubId` and `setHubID` or `setHubId` to keep up with the JS style

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we group these imports based on React, Jaia, OpenLayers, and style?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we update the return type to be an array of HTML elements? Also maybe we should add a @notes section to the comment to say would be used in a simulator with multiple hubs

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we make these `===` since we know getElementById returns null if the element is not found

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: We should make this reflect the fact the function can return null

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Let's make this `===`

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: We need the `@returns` line here

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we make this `!==`?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we update this function comment to include the return value

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: How do the boolean return values impact the `handleEvent` property?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we make this `!==`

File: src/web/jcc/client/components/SettingsPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SettingsPanel.tsx]: Ed, what do you think of wrapping the SetHubLocationPanel with the mui accoridon components, renaming that panel to HubLocationAccordion and then all we need to do is import that component in this file as one unit and call it as `<HubLocationAccordion props={...} />`

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we make the contents centered within the accordion details?

**Review by michael-jaia — CHANGES_REQUESTED** [review]

File: src/lib/messages/jaia_dccl.proto
Comment (edsanville) [reply src/lib/messages/jaia_dccl.proto]: Good point on this one, done.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: I decided to go with snake_case, only because hub_id is a field name from a protobuf message.  Our general principle in JS has been to use the snake_case only for protobuf stuff, so those fields have unambiguous names across JS, C++, and python.

@ferrom-jaia what do you think on this one?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: This one actually has no return value.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: From the OpenLayers docs:

```Method called by the map to notify the interaction that a browser event was dispatched to the map. The function may return false to prevent the propagation of the event to other interactions in the map's interactions chain.```

So, I'm preventing the click event from falling through to the other interactions on the map's interaction stack.  I added some comments to clarify the reasoning on this.

File: src/web/jcc/client/components/SettingsPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SettingsPanel.tsx]: We could do that, but then we'd need to pass `isOpenAccordionTab` and `handleAccordionTabClick` as props to the HubLocationAccordion component.  I'm not sure it's worth it, but the Accordion JSX probably does need some refactoring...  it looks like there's a lot of code duplication there.  We probably need some brainstorming to come up with a way to make it more concise and manageable.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: Yes I meant to say we should include the `@returns {void}` tag here

File: src/web/jcc/client/components/SettingsPanel.tsx
Comment (michael-jaia) [reply src/web/jcc/client/components/SettingsPanel.tsx]: Ok, I am onboard with keeping the Accordion wrapper in the settings panel. One thing I did notice when reviewing the props: what do you think of importing the instance of JaiaAPI in the file and not passing it as props?

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (michael-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Hey Ed, can we move the inline css into its own css file. A SetHubLocation.css or .less file in style/components. A less file would probably be better so we can use the nesting to avoid naming collisions with other files.

File: src/web/jcc/client/components/SettingsPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SettingsPanel.tsx]: I'm onboard with that idea.  In general, importing the api and the map, rather than passing as props would probably make sense app-wide, really.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: Ok, gotcha!

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: Yeah, we can certainly do that.  I did it in an ad-hoc way here, only because I had to finagle it to get things to center properly.  It's kind of a code smell for our css, that we need to massage the styles so much to get them to look good.  I was trying to localize those tweaks, so they didn't affect any other elements... but yes they're probably better off in a less file for that panel.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: Ok, switched to hubId

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (ferrom-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Can we use a React way to do this? This works, but it looks different than most of the jcc code with how we interact with text fields.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (ferrom-jaia) [inline src/web/jcc/client/components/SetHubLocationPanel.tsx]: Instead of SetHubLocationPanel can we use SetHubLocation for both the function name and the css name since it is not a panel

File: src/web/jcc/client/style/components/SetHubLocationPanel.less
Comment (ferrom-jaia) [inline src/web/jcc/client/style/components/SetHubLocationPanel.less]: If we are changing the name of the function this should change as well

**Review by ferrom-jaia — COMMENTED** [review]
Nice work Ed, I left a couple of comments that we should review and discuss before merging in.

**Comment by ferrom-jaia** [comment]
Can we update the pull request image or remove it since it does not match the implementation.

File: src/web/jcc/client/components/SetHubLocationPanel.tsx
Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: Ah yes...  will do this.

Comment (edsanville) [reply src/web/jcc/client/components/SetHubLocationPanel.tsx]: Ok, I used React Refs to track the two fields as uncontrolled elements.  We could use React state controlled elements, but that would be more complex code...

**Review by ferrom-jaia — APPROVED** [review]
Looks great!

**Pull request merged**
